src/script/auth/component/ProgressBar.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+
+interface ProgressProps {
+  error: boolean;
+  percent: number;
+  style?: React.CSSProperties;
+  width: number;
+}
+export const ProgressBar = ({width, percent, error, style}: ProgressProps) => {
+  const [progress, setProgress] = React.useState(0);
+
+  React.useEffect(() => setProgress(Math.floor((percent / 100) * width)), [percent, width]);
```

Comment:
You don't need a state here, since the update come only from the props, you could just do 

```
const progress = percent / 100 * width;
```